discovery-infra/tests/test_discovery_ignition_override.py

```diff
@@ -1,37 +1,159 @@
+import logging
+import copy
 import pytest
 import base64
 from tests.base_test import BaseTest
 
+TEST_STRING = "I can write tests all day long"
+
 
 class TestDiscoveryIgnition(BaseTest):
+    base64_encoded_test_string = base64.b64encode(TEST_STRING.encode("utf-8")).decode("utf8")
+    override_path = "/ignition/test_discovery_ignition_override"
+    base_ignition = {
+        "ignition": {
+            "version": "3.1.0"
+        },
+        "storage": {
+            "files": [
+                {
+                    "path": override_path,
+                    "contents": {"source": f"data:text/plain;base64,{base64_encoded_test_string}"}
+                }
+            ]
+        }
+    }
 
     @pytest.mark.regression
     def test_discovery_ignition_override(self, api_client, node_controller, cluster):
-        # Create a discovery ignition override
-        # Start the nodes
-        # Verify the override was applied
-        override_path = "/ignition/file_override"
-        test_string = "I can write tests all day long"
-        ignition_override = {"ignition": {"version": "3.1.0"}, "storage": {"files": [
-            {"path": override_path,
-             "contents": {"source": "data:text/plain;base64,{}".format(base64.b64encode(test_string))}}]}}
+        """ Test happy flow.
 
+        Create a discovery ignition override
+        Start the nodes
+        Verify the override was applied
+        """
         # Define new cluster
         cluster_id = cluster().id
+
+        ignition_override = copy.deepcopy(TestDiscoveryIgnition.base_ignition)
+        override_path = "/ignition/test_discovery_ignition_override"
+        ignition_override["storage"]["files"][0]["path"] = override_path
         api_client.patch_cluster_discovery_ignition(cluster_id, ignition_override)
+
         # Generate and download cluster ISO
         self.generate_and_download_image(cluster_id=cluster_id, api_client=api_client)
         # Boot nodes into ISO
         hosts = node_controller.start_all_nodes()
         # Wait until hosts are discovered and update host roles
         self.wait_until_hosts_are_discovered(cluster_id=cluster_id, api_client=api_client)
         # Verify override
+        self.validate_ignition_override(hosts, TestDiscoveryIgnition.override_path)
+
+    @pytest.mark.regression
+    def test_discovery_ignition_after_ISO_was_created(self, api_client, node_controller, cluster):
+        """ Verify that we create a new ISO upon discovery igniton override in case one already exists.
```

Comment:
I don't think we do this, but would probably be a good enhancement.

Reply:
I'll open a ticket for it and comment out the test